tests/integration_tests/scheduler/test_lsf_driver.py

```diff
@@ -152,3 +156,79 @@ async def finished(iens, returncode, aborted):
 
     await driver.submit(0, f"exit {actual_returncode}")
     await poll(driver, {0}, finished=finished)
+
+
+BJOBS_HEADER = (
+    "JOBID   USER    STAT  QUEUE      FROM_HOST   EXEC_HOST   JOB_NAME   SUBMIT_TIME"
+)
+
+
+@pytest.mark.parametrize(
+    "bjobs_script, expectation",
+    [
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n1 someuser DONE foo'; exit 0",
+            does_not_raise(),
+            id="all-good",
+        ),
+        pytest.param(
+            "echo 'No unfinished job found'; exit 0",
+            pytest.raises(asyncio.TimeoutError),
+            id="empty_cluster",
+        ),
+        pytest.param(
+            "echo 'Job <1> is not found' >&2; exit 0",
+            # Actual command is seen to return zero in such a scenario
+            pytest.raises(asyncio.TimeoutError),
+            id="empty_cluster_specific_id",
+        ),
+        pytest.param(
+            "echo '1 someuser DONE foo'",
+            does_not_raise(),
+            id="missing_header_is_accepted",  # (debatable)
+        ),
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n1 someuser DONE foo';"
+            " echo 'Job <2> is not found' >&2 ; exit 255",
+            # If we have some success and some failures, actual command returns with 255 (!)
+            does_not_raise(),
+            id="error_for_irrelevant_job_id",
+        ),
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n2 someuser DONE foo'",
+            pytest.raises(asyncio.TimeoutError),
+            id="wrong-job-id",
+        ),
+        pytest.param(
+            "exit 1",
+            pytest.raises(asyncio.TimeoutError),
+            id="exit-1",
+        ),
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n1 someuser DONE foo'; exit 1",
+            # (this is not observed in reality)
+            does_not_raise(),
+            id="correct_output_but_exitcode_1",
+        ),
+        pytest.param(
+            f"echo '{BJOBS_HEADER}\n1 someuser'; exit 0",
+            pytest.raises(asyncio.TimeoutError),
+            id="unparsable_output",
+        ),
+    ],
+)
+async def test_faulty_bjobs(monkeypatch, tmp_path, bjobs_script, expectation):
+    bin_path = tmp_path / "bin"
+    bin_path.mkdir()
+    monkeypatch.setenv("PATH", f"{bin_path}:{os.environ['PATH']}")
+    bsub_path = bin_path / "bsub"
+    bsub_path.write_text("#!/bin/sh\necho 'Job <1> is submitted to default queue'")
+    bsub_path.chmod(bsub_path.stat().st_mode | stat.S_IEXEC)
+    bjobs_path = bin_path / "bjobs"
+    bjobs_path.write_text(f"#!/bin/sh\n{bjobs_script}")
+    bjobs_path.chmod(bsub_path.stat().st_mode | stat.S_IEXEC)
+    driver = LsfDriver()
+    driver._poll_period = 0.01
```

Comment:
Does it even matter to set the `driver._poll_period` here?

Reply:
Nope, does not make it faster. Removed.